lms/static/js/inline_analytics.js

```diff
@@ -94,66 +97,55 @@ window.InlineAnalytics = (function() {
 
         // Build the array of choice texts
         var choiceText = getChoiceTexts(partId);
+        
+    	// Generate choice name array
+    	var choiceNameArray = convertNameToArray(choiceNameString);
 
         if (result) {
-            // Loop through results and construct row array
-            // Rows returned from the analytics API are in "option" order so we use this
-            // to determine if any rows are "missing" from the data.
-            var arrayLength = result.length;
+            // Build the array of value_id's
+            var valueIdArray = constructValueIdArray(result);
+        	
+            // Loop through choiceNameArray and construct row array.
+            // We can determine if any rows are "missing" from the api data
+            // since the choiceNameArray is a list of all possible rows.
+            var arrayLength = choiceNameArray.length;
             for (index = 0; index < arrayLength; index++) {
-
-                // valueId: option for this row in the api result
-                valueId = result[index]['value_id'];
-                // currentIndex: number of rows already added to graphic
-                currentIndex = trs.length;
-                // valueIndex: index of the option for this row in the api result
-                valueIndex = valueId.replace('choice_', '');
-
-                // If the current row we are looking to add is less than the row indicated
-                // by the option for this index then add rows for those that are missing
-                if (currentIndex < valueIndex) {
-                    insertMissingRows(partId,
-                        currentIndex,
-                        valueIndex,
-                        correctResponse,
-                        choiceText,
-                        trs);
-                }
-
-                correct = result[index]['correct'];
-                count = result[index]['count'];
-                percent = Math.round(count * 1000 / (totalAttemptCount * 10));
-
-                if (correct) {
-                    answerClass = 'inline-analytics-correct';
+            	
+                // If the choice name is not in valueIdArray then we need to add a row
+                // for the missing data.
+                if (valueIdArray.indexOf(choiceNameArray[index]) === -1) {
+                	insertMissingRows(partId,
```

Comment:
If we're going to split the params onto separate lines, how about dropping `partId` down to a new line too?